app/workers/scheduler/subscriptions_scheduler.rb

```diff
@@ -0,0 +1,12 @@
+# frozen_string_literal: true
+require 'sidekiq-scheduler'
+
+class Scheduler::SubscriptionsScheduler
+  include Sidekiq::Worker
+
+  def perform
+    Account.expiring(1.day.from_now).partitioned.pluck(:id) do |id|
```

Comment:
Maybe...


- It looks like `expiring` is only ever used for this purpose ... maybe add a method like `Account.expiring_soon` which just knows about `{ 1.day.from_now }`
- Move this whole query to a method in this class like `expiring_accounts.pluck(:id)` ...